core/core.go

```diff
@@ -41,7 +41,7 @@ type ServiceInfo struct {
 // comunicate.
 type Network interface {
 	CreateNetwork(id string) error
-	DeleteNetwork(id string) error
+	DeleteNetwork(id, encap string, pktTag, extPktTag int) error
```

Comment:
This looks a bit inconsistent, `CreateNetwork()` requires just an `id` but `DeleteNetwork()` requires a lot more info. Also by adding encap, pktTag arguments it seems like we are fixing networks to always have these attributes.

The original idea behind having a `Id` based signature was that we have a state store that can be indexed by the `Id` (or some form of it). So Create would just need Id to read the store do necessary updates and Delete would again just need an Id to read the store (this time a create runtime state) with enough info to perform the cleanup. May be that has changed now? In any case it might be a good idea to keep API signatures consistent irrespective of the approach  (i.e. state store based v/s state as API arguments)